k8s/go/cmd/resolver/resolver_test.go

```diff
@@ -48,3 +49,105 @@ val8:
 		t.Errorf("YAML walker did not visit all strings (-want +got):\n%s", diff)
 	}
 }
+
+// TestYAMLStream ensures the YAML walker correctly processes a stream of
+// YAML documents.
+func TestYAMLStream(t *testing.T) {
+	testCases := []struct {
+		name string
+		// yamlDocs is a string with one or more YAML documents.
+		yamlDocs string
+		// want is the list of strings the YAML walker should visit.
+		want []string
+		// wantDocs is the number of docs the resolver should have operated on.
+		wantDocs int
+	}{
```

Comment:
Not sure if it's a valid scenario, but can we have an empty yaml file? If so, can we have a test for that too?
```
- val1
- val2
---
---
- val3
- val4
```